tornado/web.py

```diff
@@ -1319,6 +1323,8 @@ def write_error(self, status_code: int, **kwargs: Any) -> None:
             for line in traceback.format_exception(*kwargs["exc_info"]):
                 self.write(line)
             self.finish()
+        elif self._should_not_send_content(status_code):
```

Comment:
This should be the first branch in the `if`: even in `serve_traceback` mode, it's inappropriate to send it in a response to 204/304.